contents/docs/feature-flags/bootstrapping.mdx

```diff
@@ -0,0 +1,119 @@
+---
+title: Client-side bootstrapping
+sidebar: Docs
+showTitle: true
+availability:
+    free: none
+    selfServe: full
+    enterprise: full
+---
+
+> **Note:** Bootstrapping feature flags is only available in our [JavaScript web](/docs/libraries/js) and [React Native](/docs/libraries/react-native) SDKs.
+
+Since is a delay between initializing PostHog and fetching feature flags, feature flags are not always available immediately.This can be detrimental if you want to do something like redirecting to a different page based on a feature flag.
+
+To have your feature flags available immediately, you can initialize PostHog with initial values until PostHog has had a chance to fetch them. This is called bootstrapping.
+
+To bootstrap PostHog with feature flags, use the `bootstrap` key in the initialization config:
+
+<MultiLanguage>
+
+```js-web
+posthog.init('<ph_project_api_key>', {
+    api_host: '<ph_instance_address>',
+    bootstrap: {
+        featureFlags: {
+            'flag-1': true,
+            'variant-flag': 'control',
+            'other-flag': false,
+        },
+    },
+})
+```
+
+```react-native
+// App.(js|ts)
+import { usePostHog, PostHogProvider } from 'posthog-react-native'
+
+export function MyApp() {
+    return (
+        <PostHogProvider apiKey="<ph_project_api_key>" options={{
+            host: '<ph_instance_address>',
+            bootstrap: {
+                featureFlags: {
+                    'flag-1': true,
+                    'variant-flag': 'control',
+                    'other-flag': false
+                }
+            }
+        }}>
+            <MyComponent />
+        </PostHogProvider>
+    )
+}
+```
+
+</MultiLanguage>
+
+## Setting the correct flag values
+
+To ensure you are bootstrapping PostHog with the correct flag values, we recommend fetching the flags values on your server and then passing them to your frontend.
+
+You can do this by:
+
+1. On your frontend, make a request to your server
+2. Your server fetches all the feature flags for the user by calling `getAllFlags()`
+3. Your server returns the flag values in the response
```

Comment:
This is a bit confusing to explain, but basically "frontend" request go to a server then return the frontend. Bootstrapping flag request should be handled along side the frontend request. 

```suggestion
1. When receiving a frontend request, fetch all the feature flags for the user by calling `getAllFlags()`.
2. Return the frontend request with the flags values in the response.
3. On the frontend, get the flag values from the response and add them to the `bootstrap` key in the PostHog initialization.
```